### DISCUSSION TITLE
deconstructor issue

### DISCUSSION THREAD

**Discussion by domcir** [bytedeco/javacpp#647] [Q&A] [answered]
Hi

I have a C++ wrapper class which takes as an argument a smart pointer (shared_ptr).
When I create thet object in Java I would assume that the object in the smart pointer isn't destructed when the wrapper is destructed.
The behaviour I'm observing is that the pointer object is destructed when I execute a "close()" method on the wrapper object.

Do you have any cuggestions?

**Comment by saudet** [comment, marked as answer]
If you need to create an actual shared_ptr, you'll need to wrap an appropriate factory method like make_shared and call that.

**Reply by domcir** [reply]
Let me try that, thanks.

**Reply by domcir** [reply]
The factory works, thanks again!
